### PR TITLE
Added: Save, Export, Load functionality with buttons in options-panel

### PR DESCRIPTION
**Pull request by a19matma**
Saves and/or exports a file in JSON-format.
Import only works with JSON-files as well.

### CODE REVIEW

**Review by a19steis — APPROVED** [review]
Looks good, json files can be saved/exported and imported.

**Comment by a19chrel** [comment]
Some changes need to be done, the current solution just put the loaded data into data and lines, should be added to initialState.

File: DuggaSys/diagram.js [hunk 1]
```diff
@@ -476,7 +473,12 @@ class StateMachine
         // Change the sliders max to historyLogs length
         document.getElementById("replay-range").setAttribute("max", this.historyLog.length.toString());
     }
-
+    removeFutureStates(){
+        // Remove the history entries that are after current index
+        while(this.currentHistoryIndex + 1 != this.historyLog.length) {
+            this.historyLog.pop();
+        }
```
Comment (a19hammi) [inline DuggaSys/diagram.js:480]: Wouldn't just a splice do? Same end result but doesn't need to resize the object more than once that way.

**Review by a19steis — APPROVED** [review]
Other than what hamlet commented, i dont see anything strange. Seems to be working fine.

File: DuggaSys/diagram.js (hunk 1 above)
Comment (a19chrel) [reply DuggaSys/diagram.js:480]: It is true, we just used the old code that did the removal of elements before currentIndex, do you want us to change to an splice?

Comment (a19hammi) [reply DuggaSys/diagram.js:480]: Preferably. Less code with same result.

Comment (a19chrel) [reply DuggaSys/diagram.js]: The changes are done. Can you solve the conflict and then request an review again from @a19steis

**Comment by a19hammi** [comment]
Another review my boy @a19steis

**Review by a19steis — APPROVED** [review]
Splice change looks correct, good job ! :)

**Pull request merged**